substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -2160,164 +2169,102 @@ public NativeImageDebugLocationInfo split(int stackDecrement, int adjustment) {
+     * parameter offset relative to the unadjusted callee stack pointer.
      */
-    static final int AARCH64_STACK_OFFSET = 8;
+    static final int PRE_EXTEND_FRAME_ADJUSTMENT;
 
     /**
-     * Adjustment in bytes added to frame size when recomputing parameter stack offsets after stack
-     * adjustment on AMD64.
+     * Adjustment in bytes added to raw offset for stack passed parameters on AMD64.
      *
-     * The value allows for the fact that the reported framesize includes the stacked return address
-     * which has already been rolled into the offsets derived for the unadjusted sp.
+     * The value includes a word offset from the unadjusted (pre-extend) sp to allow for the stacked
+     * return address.
      */
-    static final int AMD64_FRAMESIZE_ADJUSTMENT = -8;
+    static final int AMD64_PRE_EXTEND_FRAME_ADJUSTMENT = 8;
 
     /**
-     * Adjustment in bytes added to frame size when recomputing parameter stack offsets after stack
-     * adjustment on AARCH64.
+     * Adjustment in bytes added to raw offset for stack passed parameters on AARCH64.
      *
-     * The value in this case is zero. Although the reported framesize for AArch64 includes the
-     * pushed lr and fp registers these have not been rolled into the offsets derived for the
-     * unadjusted sp.
+     * The value is zero because nothing is pushed until the stack extend occurs.
      */
-    static final int AARCH64_FRAMESIZE_ADJUSTMENT = 0;
+    static final int AARCH64_PRE_EXTEND_FRAME_ADJUSTMENT = 0;
 
-    static int adjustFrameSize(int frameSize) {
-        // make sure this is the right arch and os
+    static {
         Architecture arch = ConfigurationValues.getTarget().arch;
         assert arch instanceof AMD64 || arch instanceof AArch64 : "unexpected architecture";
         OS os = OS.getCurrent();
         assert os == OS.LINUX || os == OS.WINDOWS : "unexpected os";
-        int adjustment = frameSize;
         if (arch instanceof AMD64) {
             // reported amd64 frame size includes an extra 8 bytes for the stacked return address
-            adjustment += AMD64_FRAMESIZE_ADJUSTMENT;
+            PRE_EXTEND_FRAME_ADJUSTMENT = AMD64_PRE_EXTEND_FRAME_ADJUSTMENT;
         } else {
-            adjustment += AARCH64_FRAMESIZE_ADJUSTMENT;
+            PRE_EXTEND_FRAME_ADJUSTMENT = AARCH64_PRE_EXTEND_FRAME_ADJUSTMENT;
         }
-        return adjustment;
     }
 
     class ParamLocationProducer {
-        Register[] gpregs;
-        Register[] fregs;
-        int nextGPRegIdx;
-        int nextFPregIdx;
-        int nextStackIdx;
-        int stackParamCount;
-        int stackOffset;
-
-        ParamLocationProducer(ResolvedJavaMethod method) {
-            Architecture arch = ConfigurationValues.getTarget().arch;
+        private final HostedMethod hostedMethod;
+        private final CallingConvention callingConvention;
+        private boolean usesStack;
+
+        private static final int adjustment;
+
+        static {
```

Comment:
Same here. The entire initialization block can be reduced to:
```java
    static {
        adjustment = ConfigurationValues.getTarget().arch.getReturnAddressSize();
    }
```

Reply:
This static field can be dropped in favour of the previous one.